seqr/views/apis/staff_api_tests.py

```diff
@@ -531,3 +606,87 @@ def test_anvil_export(self, mock_export_multiple_files):
+
+        # Test multiple dataset types error
+        mock_file_iter.return_value = SAMPLE_QC_DATA_MORE_DATA_TYPE
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'Multiple dataset types detected: wes ,wgs')
+
+        # Test unexpected data type error
+        mock_file_iter.return_value = SAMPLE_QC_DATA_UNEXPECTED_DATA_TYPE
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.reason_phrase, 'Unexpected dataset type detected: "unknown" (should be "exome" or "genome")')
+
+        # Test normal functions
+        mock_file_iter.return_value = SAMPLE_QC_DATA
+        mock_logger.reset_mock()
+        response = self.client.post(url, content_type='application/json',
+                data=json.dumps({'file': 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v15/sample_qc/final_output/seqr_sample_qc.tsv'}))
+        self.assertEqual(response.status_code, 200)
+        response_json = response.json()
+        self.assertListEqual(response_json.keys(), ['info', 'errors', 'warnings'])
+        self.assertListEqual(response_json['info'], [
+            u'Parsed 4 exome samples',
+            u'Found and updated matching seqr individuals for 3 samples'
+        ])
+        self.assertListEqual(response_json['warnings'], [
+            u'The following 1 samples were skipped: 03133B_2',
+            u'The following filter flags have no known corresponding value and were not saved: coverage_exome',
+            u'The following population platform filters have no known corresponding value and were not saved: n_deletion, n_insertion, n_snp, r_insertion_deletion'
+        ])
+
+        calls = [
+             mock.call('Parsed 4 exome samples'),
+             mock.call('Missing 1 samples from qc output'),
+             mock.call(
+                'The following filter flags have no known corresponding value and were not saved: coverage_exome'),
+             mock.call(
+                'The following population platform filters have no known corresponding value and were not saved: n_deletion, n_insertion, n_snp, r_insertion_deletion'),
+             mock.call('Found and updated matching seqr individuals for 3 samples'),
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        population = [ indiv.population for indiv in Individual.objects.filter(id__in=[1,5])]
+        population.append(Individual.objects.get(id = 2).population)
+        self.assertListEqual(population, ['NFE', 'NFE', 'AMR'])
```

Comment:
Please check for each indiviodual that they have the expected population - this is really hard to read. There are 3 edited individuals as a result of this call - for each one fetch it from the db and confirm it has the expected updates. This should also include checking filter_flags and filter_flags

Reply:
Changed to test per individual.